velox/exec/benchmarks/PrefixSortBenchmark.cpp

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <folly/Benchmark.h>
+#include <folly/init/Init.h>
+
+#include "glog/logging.h"
+#include "velox/exec/PrefixSort.h"
+#include "velox/vector/fuzzer/VectorFuzzer.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+
+namespace {
+
+// Init memoryManager
+memory::MemoryPool* rootPool() {
+  static std::shared_ptr<memory::MemoryPool> rootPool_ = []() {
+    memory::MemoryManager::initialize({});
+    return memory::memoryManager()->addRootPool("PrefixSortTest", 4L << 30);
+  }();
+  return rootPool_.get();
+}
+
+class TestCase {
+ public:
+  TestCase(
+      memory::MemoryPool* pool,
+      const std::string& testName,
+      size_t rowsNum,
+      const RowTypePtr& rowType)
+      : testName_(testName), rowsNum_(rowsNum) {
+    data_ = std::make_unique<RowContainer>(rowType->children(), pool);
+    RowVectorPtr sortedRows = makeFuzzRows(pool, rowsNum, rowType);
+    storeRows(rowsNum, sortedRows, data_.get(), rows_);
+  };
+
+  const std::string& testName() const {
+    return testName_;
+  }
+
+  const size_t rowsNum() const {
+    return rowsNum_;
+  }
+
+  const std::vector<char*>& rows() const {
+    return rows_;
+  }
+
+  RowContainer* rowContainer() const {
+    return data_.get();
+  }
+
+ private:
+  // Store sortedRows into a RowContainer data to mock the behavior of
+  // SortBuffer.
+  void storeRows(
+      const int rowsNum,
+      const RowVectorPtr& sortedRows,
+      RowContainer* data,
+      std::vector<char*>& rows) {
+    SelectivityVector allRows(rowsNum);
```

Comment:
This variable is not needed. You can use a version of DecodedVector that doesn't require SelectivityVector.

DecodedVector decoded(*sortedRows->childAt(column));